assets/js/components/PresenterStreamArea.tsx

```diff
@@ -32,19 +81,26 @@ const PresenterStreamArea = ({
     }
   }, [presenters]);
 
+  useEffect(() => {
+    if (lastSourceTypeChange != null)
+      replaceTrack(webrtc, clientName, presenterStreams[clientName], lastSourceTypeChange);
+  }, [lastSourceTypeChange]);
+
   return presenters.includes(clientName) ? (
     <>
+      <RtcClientPlayer name={clientName} clientStreamAvailable={lastSourceTypeChange!} />
       {presenters.map((presenter) => {
-        return (
-          <RtcPlayer
-            isMyself={clientName == presenter}
-            name={presenter}
-            presenterStreams={presenterStreams}
-            streamsAvailable={streamsAvailable}
-            key={presenter}
-          />
-        );
+        if (presenter != clientName)
+          return (
+            <RtcPlayer
+              isMyself={clientName == presenter}
+              name={presenter}
+              streamsAvailable={streamsAvailable}
+              key={presenter}
+            />
+          );
       })}
+      <ControlPanel clientName={clientName} setLastSourceTypeChange={setLastSourceTypeChange} />
```

Comment:
Wouldn't it be more readable to first filter presenters and then map? We could also think about storing them separately in the state?

Reply:
Filtering - fixed. "presenters" is already a state passed down from Event component as a prop.